src/Carousel/Carousel.tsx

```diff
@@ -64,90 +65,98 @@ class Carousel extends React.Component<CarouselProps, CarouselState> {
+                </li>
+              );
 
-        items.push(
-          React.cloneElement(child, {
-            key: `slider-item${index}`,
-            style: {
-              ...child.props.style,
-              [lengthKey]: `${100 / count}%`
-            },
-            className: classNames(addPrefix('slider-item'), child.props.className)
-          })
-        );
-      }
-    );
+              items.push(
+                React.cloneElement(child, {
+                  key: `slider-item${index}`,
+                  style: {
+                    ...child.props.style,
+                    [lengthKey]: `${100 / count}%`
+                  },
+                  className: classNames(addPrefix('slider-item'), child.props.className)
+                })
+              );
+            }
+          );
 
-    const activeRatio = `-${(100 / count) * active}%`;
-    const sliderStyles = {
-      [lengthKey]: `${count * 100}%`,
-      transform: vertical
-        ? `translate3d(0, ${activeRatio} ,0)`
-        : `translate3d(${activeRatio}, 0 ,0)`
-    };
+          const activeRatio = `${!vertical && context.rtl ? '' : '-'}${(100 / count) * active}%`;
+          const sliderStyles = {
+            [lengthKey]: `${count * 100}%`,
+            transform: vertical
+              ? `translate3d(0, ${activeRatio} ,0)`
+              : `translate3d(${activeRatio}, 0 ,0)`
+          };
 
-    const classes = classNames(
-      className,
-      classPrefix,
-      addPrefix(`placement-${placement}`),
-      addPrefix(`shape-${shape}`)
-    );
-    const unhandled = getUnhandledProps(Carousel, rest);
+          const classes = classNames(
+            className,
+            classPrefix,
+            addPrefix(`placement-${placement}`),
+            addPrefix(`shape-${shape}`)
+          );
+          const unhandled = getUnhandledProps(Carousel, rest);
 
-    return (
-      <Animation.Transition
-        enteredClassName={addPrefix('last')}
-        exitingClassName={addPrefix('reset')}
-        in={last}
-      >
-        <Component className={classes} {...unhandled}>
-          <div className={addPrefix('content')}>
-            <div className={addPrefix('slider')} style={sliderStyles}>
-              {items}
-            </div>
-            {count ? (
-              <div className={addPrefix('slider-after')}>{[items[items.length - 1], items[0]]}</div>
-            ) : null}
-          </div>
-          <div className={addPrefix('toolbar')}>
-            <ul>{labels}</ul>
-          </div>
-        </Component>
-      </Animation.Transition>
+          return (
+            <Animation.Transition
+              enteredClassName={addPrefix('last')}
+              exitingClassName={addPrefix('reset')}
+              in={last}
+            >
+              <Component className={classes} {...unhandled}>
+                <div className={addPrefix('content')}>
+                  <div className={addPrefix('slider')} style={sliderStyles}>
```

Comment:
The scope of the context should be as small as possible.
```jsx
<IntlContext.Consumer>
  {context => {
    const activeRatio = `${!vertical && context?.rtl ? '' : '-'}${(100 / count) * active}%`;
    const sliderStyles = {
      [lengthKey]: `${count * 100}%`,
      transform: vertical
        ? `translate3d(0, ${activeRatio} ,0)`
        : `translate3d(${activeRatio}, 0 ,0)`
    };
    return (
      <div className={addPrefix('slider')} style={sliderStyles}>
        {items}
      </div>
    );
  }}
</IntlContext.Consumer>
```

Reply:
@simonguo Done